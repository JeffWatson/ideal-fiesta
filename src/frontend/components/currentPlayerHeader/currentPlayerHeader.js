import React from 'react';
import { func, number, string } from 'prop-types';
import { contains } from 'react-immutable-proptypes';
import { connect } from 'react-redux';
import classNames from 'classnames';
import Button from 'components/controls/button';
import mapStateToProps from './selector/mapStateToProps';
import mapDispatchToProps from './actions/mapDispatchToProps';

import './currentPlayerHeader.scss';

const CurrentPlayerHeader = ({ currentPlayer, endTurn }) => {
  const username = currentPlayer.get('username');
  const money = currentPlayer.getIn(['status', 'money']);
  const specialPoints = currentPlayer.getIn(['status', 'specialPoints']);
  const className = classNames('current-player-header');

  return (<div className={className} >
    <div className="user">
      <div className="avatar" >
        <img alt={`${username} Avatar`} src={currentPlayer.get('avatarUrl')} />
      </div>
      <div className="username" >{username}</div>
    </div>
    <div className="status" >
      <div className="money">{money}</div>
      <div className="special-points">{specialPoints}</div>
      <div className="end-turn-button">
        <Button onClick={endTurn} className="end-turn-button" disabled={false} text="End Turn" />
      </div>
    </div>
  </div>);
};

CurrentPlayerHeader.propTypes = {
  currentPlayer: contains({
    username: string.isRequired,
    avatarUrl: string.isRequired,
    status: contains({
      money: number.isRequired,
      specialPoints: number.isRequired,
    }),
  }).isRequired,
  endTurn: func.isRequired,
};

export default connect(mapStateToProps, mapDispatchToProps)(CurrentPlayerHeader);
