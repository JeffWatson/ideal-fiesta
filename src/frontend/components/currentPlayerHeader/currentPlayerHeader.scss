@import "../../app/style/variables";

.current-player-header {
  padding: 1rem;
  background-color: $current-player-background;
  overflow: auto;
  display: inline-block;
  float: right;

  .user {
    width: 100px;
    height: 100px;
    border: 3px solid $white;
    text-align: center;
  }

  .user, .status {
    float: left;
    clear: none;
  }
}