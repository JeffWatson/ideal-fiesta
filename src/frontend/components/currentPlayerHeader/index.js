import ConnectedCurrentPlayerHeader from './currentPlayerHeader';

export { ConnectedCurrentPlayerHeader as default };
