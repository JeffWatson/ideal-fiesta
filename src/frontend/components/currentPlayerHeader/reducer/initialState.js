// TODO remove from here. Shouldn't exist. Move to tests only location
import { fromJS } from 'immutable';

export default fromJS({
  enabled: false,
  currentPlayer: {
    username: 'bananaMan',
    avatarUrl: './favicon.ico',
    status: {
      money: 1000,
      specialPoints: 90,
    },
  },
});
