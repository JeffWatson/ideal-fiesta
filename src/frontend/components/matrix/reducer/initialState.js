import { fromJS } from 'immutable';


// TODO need to refactor units... they should have health. Buildings too.
const initialState = {
  matrix: {
    columns: 5,
    rows: 5,
    grid: [
      [{ terrain: 'SAND', unit: 'SOLDIER', health: 20 },
        { terrain: 'SAND' },
        { terrain: 'SAND' },
        { terrain: 'SAND', unit: 'SMALL_TANK', health: 20 },
        { terrain: 'SAND' }],
      [{ terrain: 'OCEAN' },
        { terrain: 'OCEAN' },
        { terrain: 'OCEAN' },
        { terrain: 'OCEAN' },
        { terrain: 'LAND_FACTORY', health: 20 }],
      [{ terrain: 'GRASS', unit: 'SOLDIER', health: 20 },
        { terrain: 'SAND' },
        { terrain: 'MOUNTAIN' },
        { terrain: 'SAND' },
        { terrain: 'SAND' }],
      [{ terrain: 'SAND' },
        { terrain: 'SAND' },
        { terrain: 'SAND' },
        { terrain: 'GRASS', unit: 'SMALL_TANK', health: 20 },
        { terrain: 'SAND' }],
      [{ terrain: 'SAND' },
        { terrain: 'GRASS' },
        { terrain: 'SAND' },
        { terrain: 'SAND' },
        { terrain: 'SAND', unit: 'SOLDIER', health: 20 }],
    ],
  },
};

export default fromJS(initialState);
