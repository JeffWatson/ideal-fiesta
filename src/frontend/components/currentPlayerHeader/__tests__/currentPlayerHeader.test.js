import React from 'react';
import { shallow } from 'enzyme';
import { CurrentPlayerHeader } from './../currentPlayerHeader';
import initialState from './../reducer/initialState';

const props = {
  currentPlayer: initialState.get('currentPlayer'),
  endTurn: () => {},
};

describe('frontend/components/CurrentPlayerHeader component', () => {
  it('renders correctly', () => {
    const component = shallow(<CurrentPlayerHeader {...props} />);

    expect(component).toMatchSnapshot();
  });
});
