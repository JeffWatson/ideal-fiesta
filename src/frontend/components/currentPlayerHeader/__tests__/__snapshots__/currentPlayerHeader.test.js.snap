// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`frontend/components/CurrentPlayerHeader component renders correctly 1`] = `
<div
  className="current-player-header"
>
  <div
    className="user"
  >
    <div
      className="avatar"
    >
      <img
        alt="bananaMan Avatar"
        src="./favicon.ico"
      />
    </div>
    <div
      className="username"
    >
      bananaMan
    </div>
  </div>
  <div
    className="status"
  >
    <div
      className="money"
    >
      1000
    </div>
    <div
      className="special-points"
    >
      90
    </div>
    <div
      className="end-turn-button"
    >
      <Button
        className="end-turn-button"
        disabled={false}
        onClick={[Function]}
        text="End Turn"
      />
    </div>
  </div>
</div>
`;
